apollo-integration/src/test/java/com/apollographql/apollo/Rx2ApolloTest.kt

```diff
@@ -0,0 +1,256 @@
+        .assertValueAt(0) { data ->
+          assertThat(data?.hero()?.name()).isEqualTo("R2-D2")
+          true
+        }
+        .assertValueAt(1) { data ->
+          assertThat(data?.hero()?.name()).isEqualTo("Artoo")
+          true
+        }
+  }
+
+  @Test
+  @Throws(Exception::class)
+  fun queryWatcherNotCalledWhenCanceled() {
+    server.enqueue(mockResponse(FILE_EPISODE_HERO_NAME_WITH_ID))
+    val testObserver: TestObserver<EpisodeHeroNameQuery.Data> = TestObserver<EpisodeHeroNameQuery.Data>()
+    val scheduler = TestScheduler()
+    val disposable: Disposable = Rx2Apollo
+        .from(apolloClient.query(EpisodeHeroNameQuery(Input.fromNullable(EMPIRE))).watcher())
+        .map({ response -> response.data() })
+        .observeOn(scheduler)
+        .subscribeWith(testObserver)
+    scheduler.triggerActions()
+    server.enqueue(mockResponse(FILE_EPISODE_HERO_NAME_CHANGE))
+    apolloClient.query(EpisodeHeroNameQuery(Input.fromNullable(EMPIRE))).responseFetcher(NETWORK_ONLY)
+        .enqueue(null)
+    disposable.dispose()
+    scheduler.triggerActions()
+    testObserver
+        .assertValueCount(1)
+        .assertValueAt(0) { data ->
+          assertThat(data?.hero()?.name()).isEqualTo("R2-D2")
+          true
+        }
+  }
+
+  @Test
+  fun disposingAnOperationDoesNotThrowUndeliverableException() {
+    /*
+     * A simple cache that will always throw errors
+     */
+    val cacheFactory = object: NormalizedCacheFactory<NormalizedCache>() {
+      override fun create(recordFieldAdapter: RecordFieldJsonAdapter): NormalizedCache {
+        return object: NormalizedCache() {
+          override fun clearAll() {
+            throw Exception("not implemented")
+          }
+
+          override fun loadRecord(key: String, cacheHeaders: CacheHeaders): Record? {
+            throw Exception("not implemented")
+          }
+
+          override fun performMerge(apolloRecord: Record, oldRecord: Record?, cacheHeaders: CacheHeaders): Set<String> {
+            throw Exception("not implemented")
+          }
+
+          override fun remove(cacheKey: CacheKey, cascade: Boolean): Boolean {
+            throw Exception("not implemented")
+          }
+        }
+      }
+    }
+
+    val apolloClient = ApolloClient.builder()
+        .normalizedCache(cacheFactory)
+        .serverUrl("https://unused/")
+        .build()
+
+
+    val savedHandler = RxJavaPlugins.getErrorHandler()
+
+    var undeliverableException: Throwable? = null
+    RxJavaPlugins.setErrorHandler {
+      undeliverableException = it
+    }
+
+    val operation = apolloClient.apolloStore.write(EpisodeHeroNameQuery(Input.fromNullable(EMPIRE)), EpisodeHeroNameQuery.Data(EpisodeHeroNameQuery.Hero("", "")))
+    val testObserver = Rx2Apollo.from(operation)
+        .test()
+
+    testObserver.dispose()
+
+    // Since there is no cancellation mechanism for the ApolloStoreOperation, the only way to see if an error is thrown is to wait here
+    Thread.sleep(200)
```

Comment:
I've also used this approach to verify no error has been thrown.
https://bryanherbst.com/2020/07/15/rxjava-uncaught-exception-tests/